setup-slave.sh

```diff
@@ -14,7 +14,7 @@ source ec2-variables.sh
 
 # Set hostname based on EC2 private DNS name, so that it is set correctly
 # even if the instance is restarted with a different private DNS name
-PRIVATE_DNS=`wget -q -O - http://instance-data.ec2.internal/latest/meta-data/local-hostname`
+PRIVATE_DNS=`wget -q -O - http://169.254.169.254/latest/meta-data/local-hostname`
```

Comment:
We need to invoke the new setup-hostname script here -- You can assume that the spark-ec2 directory exists, so just adding a line like `bash /root/spark-ec2/setup-hostname.sh` should be sufficient